src/test/txvalidationcache_tests.cpp

```diff
@@ -5,23 +5,31 @@
 #include <consensus/validation.h>
 #include <key.h>
 #include <random.h>
+#include <script/sigcache.h>
 #include <script/sign.h>
 #include <script/signingprovider.h>
 #include <test/util/setup_common.h>
 #include <txmempool.h>
 #include <util/chaintype.h>
+#include <util/check.h>
 #include <validation.h>
 
 #include <boost/test/unit_test.hpp>
 
 struct Dersig100Setup : public TestChain100Setup {
+    ValidationCache m_validation_cache;
```

Comment:
I was thinking it might be useful to control the caches directly for the tests, but on second thought that does not seem like a good thing. Will revert.